### DISCUSSION TITLE
Seassan Future

### DISCUSSION THREAD

**Discussion by mikhi32** [seasann/seasann#21] [Announcements]
TLDR; Seassan and all connected repos's will be (again) deprecated. Please use my new repleacment [setso](https://github.com/setsojs/setso).

# Seassan history and future

Seassan was a side project that I started in the summer of 2022. I was bored, and figured: "Can I build something that compiles markdown to html and serve's it using express". Nowadays this would sound crazy in my head, but at the time it seemed like a great idea, and something "Nobody Has ever Thought about".

## Why Seassan?

Finding NPM package names is impossible. Everybody knows that. I had recently read that vite meant fast in French. So following that, I tried a bunch of different languages. I don't remember the word, but I do remember that it was irish.

## First versions

After that, I started working and, contrary to what I thought, I actually finished the project. I published it, and worked on it. I also got my first contributors. Unfortunately, a mix of not having time, and just not wanting to work (and also the code being a mess, because I used callback). I felt that js was really not making it. So...

## The first deprecation

In a quick rage I "deprecated" it and all connected projects, archiving the repository and completely cancelling all planned features.

## A new project

But, I really liked the idea tough. But I did not want to touch any js. So I started heaterdom. It was made I python and was actually pretty good, as I was way more experienced in the snake language. I realized that using a backend framework to serve simple html files, was useless, and I just integrated a normal http server. 

## A slime of hope.

There is a thing tough, with my projects, that, when I finished them, that I don't want to touch them, and I just want to start something new. But I really wanted to port all the hard work I had done with heaterdom, back to seassan. I worked hard, and with the new async/await strategies I was using, it also fixed some strange bugs. I also made it work everywhere, because for some reason I was using curl with shell commands.

## A fresh new start

After cleaning up this dying project I again, stopped working on it. With the not so much spare time I had I didn't want to use it to work on a project I felt was really just dying.  So, I started a new repo. Just a test, and I slowly developed it. In the end the idea to deprecate seassan again was popping up in my head, but I didn't have time to write something professional, because I wanted a professional goodbye, not something quick and bad.

## What now?

Going forward, seassan will be deprecated and this and all related repos will be archived. If you want a new, cleaner and more minimal alternatives, that supports the majority of things that seassan supports but with better documentation and a configuration system, then use [setso](https://github.com/setsojs/setso).

If you came this far, thanks for reading. Ive changed a lot, and the evolution of my projects really symbolizes that. 

Thanks!

Micziz
